nodeup/pkg/model/networking/canal.go

```diff
@@ -0,0 +1,71 @@
+/*
+Copyright 2020 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package networking
+
+import (
+	"k8s.io/klog"
+	"k8s.io/kops/nodeup/pkg/model"
+	"k8s.io/kops/pkg/systemd"
+	"k8s.io/kops/upup/pkg/fi"
+	"k8s.io/kops/upup/pkg/fi/nodeup/nodetasks"
+)
+
+// CanalBuilder writes Flannel's assets
+type CanalBuilder struct {
+	*model.NodeupModelContext
+}
+
+var _ fi.ModelBuilder = &CanalBuilder{}
+
+// Build is responsible for configuring the network cni
+func (b *CanalBuilder) Build(c *fi.ModelBuilderContext) error {
+	networking := b.Cluster.Spec.Networking
+
+	if networking.Canal == nil {
+		return nil
+	}
+
+	b.AddCNIBinAssets(c, []string{"flannel"})
+	b.buildFlannelTxChecksumOffloadDisableService(c)
+
+	return nil
+}
+
+// Tx checksum offloading is buggy for NAT-ed VXLAN endpoints, leading to an invalid checksum sent and causing
+// Flannel to stop to working as the traffic is being discarded by the receiver.
+// https://github.com/coreos/flannel/issues/1279
+func (b *CanalBuilder) buildFlannelTxChecksumOffloadDisableService(c *fi.ModelBuilderContext) {
```

Comment:
Is this one copied between Canal and Flannel now? 
Since `*CanalBuilder` isn't used here, you could just make this a helper function rather than to make it part of the `CanalBuilder` type.

Reply:
Good idea. Done.